### PR TITLE
SNOW-1368544: Add frontend methods for expanding window

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1368544

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   This PR adds frontend methods for expanding window.

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/base.py [hunk 1]
```diff
@@ -1735,16 +1735,38 @@ def ewm(
...
         )
 
-    @base_not_implemented()
     def expanding(
         self, min_periods=1, axis=0, method="single"
     ):  # noqa: PR01, RT01, D200
         """
         Provide expanding window calculations.
         """
-        # TODO: SNOW-1119855: Modin upgrade - modin.pandas.base.BasePandasDataset
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/base.py:1745]: This ticket is not done yet. Can we keep the todo?

File: src/snowflake/snowpark/modin/pandas/window.py [hunk 2]
```diff
@@ -199,18 +199,6 @@ def count(self, numeric_only: bool = False):
         # TODO: SNOW-1063358: Modin upgrade - modin.pandas.window.Rolling
         return self._aggregate(method_name="count", numeric_only=numeric_only)
 
-    def sem(
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/pandas/window.py:202]: Why was this moved down?

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 3]
```diff
@@ -10543,17 +10543,6 @@ def rolling_count(
     ) -> None:
         ErrorMessage.method_not_implemented_error(name="count", class_="Rolling")
 
-    def rolling_sem(
```
Comment (sfc-gh-helmeleegy) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:10546]: Same, not clear why this was moved down.

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Looks good overall. Just a couple nits/questions. Thanks for starting on this work!

File: src/snowflake/snowpark/modin/pandas/base.py (hunk 1 above)
Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/modin/pandas/base.py:1745]: This TODO should be done even though the overall ticket may not be, the code below for expanding is a straight copy-paste from Modin

File: src/snowflake/snowpark/modin/pandas/window.py (hunk 2 above)
Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/modin/pandas/window.py:202]: Moved this down just to match pandas order of aggregations in their documentation: https://pandas.pydata.org/docs/reference/window.html

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 3 above)
Comment (sfc-gh-nkrishna) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:10546]: Ditto above

**Review by sfc-gh-joshi — APPROVED** [review]
LGTM. If you have some time, can you make a PR to upstream modin to update the signatures of expanding methods to avoid *args/**kwargs where possible? Afterwards we'll be able to use the upstream version of the class without making docs look strange.

For what it's worth, if we start using the current upstream version of the class + use the docstring plugin (even with *args/**kwargs instead of properly enumerated parameters), I think the inaccurate method stub would only show up in the method definition at the top of its documentation page. As long as our docstrings our correct, the description for individual parameters should be accurate.
